### DISCUSSION TITLE
Welcome to Lettuce Discussions!

### DISCUSSION THREAD

**Discussion by mp911de** [redis/lettuce#1554] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by jigneshmpatel** [comment]
I have tried use Redis 6 in context of AWS elasticache.
This is my sample code.
public class ConnectToRedisCluster {
    public static void main(String[] args) {

        String authPassword = "password";
        char charArray[] = authPassword.toCharArray();
        RedisURI redisURI = RedisURI.Builder.redis(redisDetails.getHost(), redisDetails.getPort()).withPassword(charArray).withSsl(true).build();

        //if(properties.getIsRedisEncrypted()){
           // redisURI.setSsl(Boolean.TRUE);
            //redisURI.setStartTls(Boolean.TRUE);
        //}
        RedisClusterClient clusterClient = RedisClusterClient.create(redisURI);
        StatefulRedisClusterConnection<String, String> connection = clusterClient.connect();
        RedisAdvancedClusterCommands<String, String> syncCommands = connection.sync();


        System.out.println("Connected to Redis");

        connection.close();
        clusterClient.shutdown();
    }
}

It has encryption in transit and at rest enabled and used with AUTH "default "user. The code seems working at this time, thanks to help from Mark Paluch. I will keep updating this thread, if there is more to share.

**Reply by mp911de** [reply]
Please create a new post in the Q&A section.
